src/uct/ib/base/ib_iface.c

```diff
@@ -889,10 +893,7 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
     }
 
     /* Address scope and size */
-    if (uct_ib_iface_is_roce(self) || config->is_global ||
-        /* check ADDR_TYPE for backward compatibility */
-        (config->addr_type == UCT_IB_ADDRESS_TYPE_SITE_LOCAL) ||
-        (config->addr_type == UCT_IB_ADDRESS_TYPE_GLOBAL) ||
+    if (uct_ib_iface_is_roce(self) || self->config.is_global ||
```

Comment:
i'd include `uct_ib_iface_is_roce(self)` to the definition of `self->config.is_global` as well. So you will not need to check it 
in `uct_ib_iface_fill_ah_attr_from_gid_lid`

Reply:
done